cadc-conesearch/src/main/java/org/opencadc/conesearch/TAPQueryGenerator.java

```diff
@@ -0,0 +1,236 @@
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.conesearch;
+
+import ca.nrc.cadc.dali.Circle;
+import ca.nrc.cadc.dali.CommonParamValidator;
+import ca.nrc.cadc.dali.tables.votable.VOTableWriter;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * Class to produce an ADQL query from a set of Parameters
+ */
+public class TAPQueryGenerator extends CommonParamValidator {
+    private static final int DEF_MAXREC = 1000;
+    private static final int MAX_MAXREC = Integer.MAX_VALUE;
+    private static final int MIN_VERB_VALUE = 1;
+    private static final int MID_VERB_VALUE = 2;
+    private static final int MAX_VERB_VALUE = 3;
+    private static final Logger LOGGER = LogManager.getLogger(TAPQueryGenerator.class);
+
+    protected final String catalog;
+    protected final String lowVerbositySelectList;
+    protected final String midVerbositySelectList;
+    protected final String highVerbositySelectList;
+
+    public TAPQueryGenerator(final String catalog, final String lowVerbositySelectList,
+                             final String midVerbositySelectList, final String highVerbositySelectList) {
+        this.catalog = catalog;
+        this.lowVerbositySelectList = lowVerbositySelectList;
+        this.midVerbositySelectList = midVerbositySelectList;
+        this.highVerbositySelectList = highVerbositySelectList;
+    }
+
+    /**
+     * Map with the REQUEST, LANG, and QUERY parameters.
+     *
+     * @param positionColumnName The name of the positional column to compare against the cone's position,
+     *                           preferably spatially indexed.
+     * @return map of parameter names and values
+     */
+    public Map<String, Object> getParameterMap(final String positionColumnName,
+                                               final Map<String, List<String>> parameters) {
+        final Map<String, Object> queryParameterMap = new HashMap<>();
+
+        // Obtain and, if necessary, provide a default RESPONSEFORMAT.
+        final String requestedResponseFormat = getFirstParameter(ParameterNames.RESPONSEFORMAT.name(), parameters,
```

Comment:
ConeSearch allows for output format other than VOTable? 

If so, use `CommonParamValidator.getResponseFormat(...)`

Reply:
It does as of 1.1:
https://www.ivoa.net/documents/ConeSearch/20200828/WD-ConeSearch-1.1-20200828.html#tth_sEc2.1.6